external-crates/move/move-compiler/src/typing/dependency_ordering.rs

```diff
@@ -293,52 +299,46 @@ fn module(context: &mut Context, mident: ModuleIdent, mdef: &E::ModuleDefinition
 // Scripts cannot affect the dependency graph because 1) a script cannot friend anything and 2)
 // nothing can depends on a script. Therefore, we iterate over the scripts just to collect their
 // immediate dependencies.
-fn script_defs(context: &mut Context, scripts: &BTreeMap<Symbol, E::Script>) {
+fn script_defs(context: &mut Context, scripts: &BTreeMap<Symbol, T::Script>) {
     scripts
         .iter()
         .for_each(|(sname, sdef)| script(context, *sname, sdef))
 }
 
-fn script(context: &mut Context, sname: Symbol, sdef: &E::Script) {
+fn script(context: &mut Context, sname: Symbol, sdef: &T::Script) {
     context.current_node = Some(NodeIdent::Script(sname));
     function(context, &sdef.function);
-    sdef.specs
-        .iter()
-        .for_each(|sblock| spec_block(context, sblock));
+    for (mident, sp!(loc, neighbor_)) in &sdef.spec_dependencies {
+        let dep = match neighbor_ {
+            Neighbor_::Dependency => DepType::Use,
+            Neighbor_::Friend => DepType::Friend,
+        };
+        context.add_neighbor(*mident, dep, *loc);
+    }
 }
 
 //**************************************************************************************************
 // Function
 //**************************************************************************************************
 
-fn function(context: &mut Context, fdef: &E::Function) {
+fn function(context: &mut Context, fdef: &T::Function) {
     function_signature(context, &fdef.signature);
-    function_acquires(context, &fdef.acquires);
-    if let E::FunctionBody_::Defined(seq) = &fdef.body.value {
+    if let T::FunctionBody_::Defined(seq) = &fdef.body.value {
         sequence(context, seq)
     }
-    fdef.specs
-        .values()
-        .for_each(|sblock| spec_block(context, sblock));
 }
 
-fn function_signature(context: &mut Context, sig: &E::FunctionSignature) {
+fn function_signature(context: &mut Context, sig: &N::FunctionSignature) {
     types(context, sig.parameters.iter().map(|(_, st)| st));
     type_(context, &sig.return_type)
 }
 
-fn function_acquires(context: &mut Context, acqs: &[E::ModuleAccess]) {
```

Comment:
Is this going away because this is Sui and we don't use `acquires`, or because acquires aren't a thing at this stage in the compiler pipeline? (or should this not be deleted at all)?

Reply:
Because acquires are always local to the module that uses them. So any neighbor here an error.